core/TempPath.php

```diff
@@ -11,8 +11,9 @@
 function getTempFolder($base = null) {
 	$parent = getTempParentFolder($base);
 
-	// The actual temp folder is a subfolder of getTempParentFolder(), named by username
-	$subfolder = $parent . DIRECTORY_SEPARATOR . getTempFolderUsername();
+	// The actual temp folder is a subfolder of getTempParentFolder(), named by username and suffixed with currently used php-version
+	$phpversion = '-php' . preg_replace('/[^\w-\.+]+/', '-', PHP_VERSION);
+	$subfolder = $parent . DIRECTORY_SEPARATOR . getTempFolderUsername() . $phpversion;
```

Comment:
@sminnee - done. I promise 'll run test locally next time. Sorry for the commit-mess.